osrdyne/src/api.rs

```diff
@@ -43,11 +66,41 @@ pub async fn create_server(
 
 async fn app_state_updater(
     state: AppState,
-    mut known_workers: tokio::sync::watch::Receiver<Arc<Vec<WorkerMetadata>>>,
+    mut known_workers_recv: watch::Receiver<Arc<Vec<WorkerMetadata>>>,
+    mut worker_status_recv: watch::Receiver<Arc<HashMap<Vec<u8>, WorkerStatus>>>,
 ) {
```

Comment:
It feels like a lot of things will have to be adapted once multiple workers could be running for one infra.

Reply:
This is the next step indeed